pkgs/development/tools/marksman/update.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=./. -i bash -p curl jq common-updater-scripts
+#shellcheck shell=bash
+
+set -eu -o pipefail
+shopt -s inherit_errexit
+
+version=$(curl -s https://api.github.com/repos/artempyanykh/marksman/releases/latest | jq -r .tag_name)
```

Comment:
This should optionally take a GITHUB_TOKEN

```suggestion
version=$(curl -s ${GITHUB_TOKEN:+"-u \":$GITHUB_TOKEN\""} https://api.github.com/repos/artempyanykh/marksman/releases/latest | jq -r .tag_name)
```

Reply:
Just a note. I noticed this `${GITHUB_TOKEN:+"-u \":$GITHUB_TOKEN\""}` in some places in nixpkgs, but I think it's just wrong. This substitution expands to one word: `-u ":$GITHUB_TOKEN"`. Correct substitution `${GITHUB_TOKEN:+-u ":$GITHUB_TOKEN"}` expands to two words: `-u` and `:$GITHUB_TOKEN`.

You can test it:

```bash
stas@server2:~$ a="aa bb"
stas@server2:~$ set -x
stas@server2:~$ : ${a:+"-u \":$a\""}
+ : '-u ":aa bb"'
stas@server2:~$ : ${a:+-u ":$a"}
+ : -u ':aa bb'
```

In essence with `${GITHUB_TOKEN:+"-u \":$GITHUB_TOKEN\""}` curl will use username ` "` (space and quote) and password `$GITHUB_TOKEN"` (with quote at the end). Token will be invalid.

Reply:
Can you fix this up, please?

``GITHUB_API_TOKEN`` is just another alias for ``GITHUB_TOKEN`` and probably the 5th I have in my configuration.

Reply:
I just did a double check and it seems to expand correct. I think we are missing an outer quote, so it is resplit.

Reply:
> You still think that `${GITHUB_TOKEN:+"-u \":$GITHUB_TOKEN\""}` is correct?

No, but it works probably due to resplicing.

Reply:
#204219